tools/pythonpkg/tests/fast/test_type.py

```diff
@@ -0,0 +1,124 @@
+        assert str(UTINYINT) == 'UTINYINT'
+        assert str(SMALLINT) == 'SMALLINT'
+        assert str(USMALLINT) == 'USMALLINT'
+        assert str(INTEGER) == 'INTEGER'
+        assert str(UINTEGER) == 'UINTEGER'
+        assert str(BIGINT) == 'BIGINT'
+        assert str(UBIGINT) == 'UBIGINT'
+        assert str(HUGEINT) == 'HUGEINT'
+        assert str(UUID) == 'UUID'
+        assert str(FLOAT) == 'FLOAT'
+        assert str(DOUBLE) == 'DOUBLE'
+        assert str(DATE) == 'DATE'
+        assert str(TIMESTAMP) == 'TIMESTAMP'
+        assert str(TIMESTAMP_MS) == 'TIMESTAMP_MS'
+        assert str(TIMESTAMP_NS) == 'TIMESTAMP_NS'
+        assert str(TIMESTAMP_S) == 'TIMESTAMP_S'
+        assert str(TIME) == 'TIME'
+        assert str(TIME_TZ) == 'TIME WITH TIME ZONE'
+        assert str(TIMESTAMP_TZ) == 'TIMESTAMP WITH TIME ZONE'
+        assert str(VARCHAR) == 'VARCHAR'
+        assert str(BLOB) == 'BLOB'
+        assert str(BIT) == 'BIT'
+        assert str(INTERVAL) == 'INTERVAL'
+
+    def test_array_type(self):
+        type = duckdb.array_type(BIGINT)
+        assert str(type) == 'BIGINT[]'
+
+    def test_struct_type(self):
+        type = duckdb.struct_type({'a': BIGINT, 'b': BOOLEAN})
+        assert str(type) == 'STRUCT(a BIGINT, b BOOLEAN)'
+
+        type = duckdb.struct_type([BIGINT, BOOLEAN])
+        assert str(type) == 'STRUCT(v1 BIGINT, v2 BOOLEAN)'
+
+    def test_map_type(self):
+        type = duckdb.map_type(duckdb.sqltype("BIGINT"), duckdb.sqltype("DECIMAL(10, 2)"))
+        assert str(type) == 'MAP(BIGINT, DECIMAL(10,2))'
+
+    def test_decimal_type(self):
+        type = duckdb.decimal_type(5, 3)
+        assert str(type) == 'DECIMAL(5,3)'
+
+    def test_string_type(self):
+        type = duckdb.string_type()
+        assert str(type) == 'VARCHAR'
+
+    def test_string_type_collation(self):
+        type = duckdb.string_type('NOCASE')
+        # collation does not show up in the string representation..
+        assert str(type) == 'VARCHAR'
+
+    def test_union_type(self):
+        type = duckdb.union_type([BIGINT, VARCHAR, TINYINT])
+        assert str(type) == 'UNION(v1 BIGINT, v2 VARCHAR, v3 TINYINT)'
+
+        type = duckdb.union_type({'a': BIGINT, 'b': VARCHAR, 'c': TINYINT})
+        assert str(type) == 'UNION(a BIGINT, b VARCHAR, c TINYINT)'
+    
+    import sys
+    @pytest.mark.skipif(sys.version_info < (3,8), reason="requires >=python3.8")
+    def test_implicit_convert_from_builtin_type(self):
+        type = duckdb.list_type(list[str])
+        assert str(type.child) == "VARCHAR[]"
+
+        mapping = {
+            'VARCHAR': str,
+            'BIGINT': int,
+            'BLOB': bytes,
+            'BLOB': bytearray,
+            'BOOLEAN': bool,
+            'DOUBLE': float
+        }
+        for expected, type in mapping.items():
+            res = duckdb.list_type(type)
+            assert str(res.child) == expected
+        
+        res = duckdb.list_type(dict['a': str, 'b': int])
```

Comment:
Having this function called `list_type` seems a bit off, can we name this function `to_duck_type` if it is a function that takes an arbitrary Python object and converts it into a DuckDB type?

Reply:
This only showcasing that wherever a DuckDBPyType is accepted, we also implicitly accept a python builtin type - and convert it to a DuckDBPyType

`duckdb.map_type(list[str], bool)` would also work

Reply:
Ah, that makes sense. Thanks